server/src/middleware/host_filter.rs

```diff
@@ -145,14 +161,14 @@ where
 impl WhitelistedHosts {
 	fn recognize(&self, other: &Authority) -> bool {
 		if let Ok(p) = self.0.recognize(&other.host) {
-			let p = p.handler();
+			let ports = p.handler();
 
-			match (p, &other.port) {
+			ports.iter().any(|p| match (p, &other.port) {
```

Comment:
I expect this list to be small and linear search should ok here, we could change this to have HashSet or BTreeSet though

Reply:
I suspect you're right!